thrust/thrust/detail/type_traits.h

```diff
@@ -47,244 +47,24 @@ using integral_constant = ::cuda::std::integral_constant<T, v>;
 
-// template<typename T> struct is_integral : public std::tr1::is_integral<T> {};
-template <typename T>
-struct is_integral : public false_type
-{};
-template <>
-struct is_integral<bool> : public true_type
-{};
-template <>
-struct is_integral<char> : public true_type
-{};
-template <>
-struct is_integral<signed char> : public true_type
-{};
-template <>
-struct is_integral<unsigned char> : public true_type
-{};
-template <>
-struct is_integral<short> : public true_type
-{};
-template <>
-struct is_integral<unsigned short> : public true_type
-{};
-template <>
-struct is_integral<int> : public true_type
-{};
-template <>
-struct is_integral<unsigned int> : public true_type
-{};
-template <>
-struct is_integral<long> : public true_type
-{};
-template <>
-struct is_integral<unsigned long> : public true_type
-{};
-template <>
-struct is_integral<long long> : public true_type
-{};
-template <>
-struct is_integral<unsigned long long> : public true_type
-{};
-template <>
-struct is_integral<const bool> : public true_type
-{};
-template <>
-struct is_integral<const char> : public true_type
-{};
-template <>
-struct is_integral<const unsigned char> : public true_type
-{};
-template <>
-struct is_integral<const short> : public true_type
-{};
-template <>
-struct is_integral<const unsigned short> : public true_type
-{};
-template <>
-struct is_integral<const int> : public true_type
-{};
-template <>
-struct is_integral<const unsigned int> : public true_type
-{};
-template <>
-struct is_integral<const long> : public true_type
-{};
-template <>
-struct is_integral<const unsigned long> : public true_type
-{};
-template <>
-struct is_integral<const long long> : public true_type
-{};
-template <>
-struct is_integral<const unsigned long long> : public true_type
-{};
-
-template <typename T>
-struct is_floating_point : public false_type
-{};
-template <>
-struct is_floating_point<float> : public true_type
-{};
-template <>
-struct is_floating_point<double> : public true_type
-{};
-template <>
-struct is_floating_point<long double> : public true_type
-{};
-
-template <typename T>
-struct is_arithmetic : public is_integral<T>
-{};
-template <>
-struct is_arithmetic<float> : public true_type
-{};
-template <>
-struct is_arithmetic<double> : public true_type
-{};
-template <>
-struct is_arithmetic<const float> : public true_type
-{};
-template <>
-struct is_arithmetic<const double> : public true_type
-{};
-
-template <typename T>
-struct is_pointer : public false_type
-{};
-template <typename T>
-struct is_pointer<T*> : public true_type
-{};
-
 template <typename T>
 struct is_device_ptr : public false_type
 {};
 
 template <typename T>
-struct is_void : public false_type
-{};
-template <>
-struct is_void<void> : public true_type
-{};
-template <>
-struct is_void<const void> : public true_type
-{};
-
-template <typename T>
-struct is_non_bool_integral : public is_integral<T>
+struct is_non_bool_integral : public ::cuda::std::is_integral<T>
```

Comment:
Suggestion: The `public` is not necessary.
```suggestion
struct is_non_bool_integral : ::cuda::std::is_integral<T>
```
One more occurrence below.